samza-core/src/main/java/org/apache/samza/job/model/ContainerModel.java

```diff
@@ -37,32 +37,48 @@
  * </p>
  */
 public class ContainerModel implements Comparable<ContainerModel> {
+  @Deprecated
   private final int containerId;
+  private final String processorId;
   private final Map<TaskName, TaskModel> tasks;
 
-  public ContainerModel(int containerId, Map<TaskName, TaskModel> tasks) {
+  public ContainerModel(String containerId, Map<TaskName, TaskModel> tasks) {
+    this(containerId, Integer.parseInt(containerId), tasks);
+  }
+
+  public ContainerModel(String processorId, int containerId, Map<TaskName, TaskModel> tasks) {
```

Comment:
Yeah.. I wasn't sure whether to annotate with deprecated because I introduced only now :) 
We should be deprecating this after 0.13, though.